scality_glance_store/store.py

```diff
@@ -117,6 +118,49 @@ def __init__(self, conf):
     def get_schemes():
         return (SCALITY_SCHEME,)
 
+    def _get_range_headers(self, image_id, offset=0, chunk_size=None):
+        """
+        Calculate range headers for partial object retrieval.
+        """
+        # Obtain object size by a HEAD request. This is to ensure compatibiliy
+        # with versions of sproxyd which has limited support for implicit
+        # ranges in the HTTP Range header.
+        size = 0
+        try:
+            headers = self._sproxyd_client.head(image_id)
```

Comment:
After taking another look at this, I realised that the `size` is used for bounds checking just before returning the headers. Moving this block makes sense as an optimization, but will not allow for range validation in all cases.